### PR TITLE
Advisor-2061: system details empty states are refactored

### PR DESCRIPTION
**Pull request by mkholjuraev**
Resolves: [Advisor](https://issues.redhat.com/browse/ADVISOR-2061).

To test:

1. Visit the system details page in Advisor application.
2. Filter the table to get empty state, check if everything is ok.
3. Open system details page with 0 recomendations, check if everything is ok.

### CODE REVIEW

**Comment by johnsonm325** [comment]
In case you don't know, you'll just need to remove two spaces per line in the EmptyStates.js file.

**Review by johnsonm325 — CHANGES_REQUESTED** [review]
I am having trouble testing this because I don't know how to get a system with no recommendations. If you can help me with that, then I can test this fully.

File: src/SmartComponents/SystemAdvisor/SystemAdvisor.js [hunk 1]
```diff
@@ -165,16 +161,13 @@ const BaseSystemAdvisor = () => {
...
     const builtRows = activeReports.flatMap((value, key) => {
       const rule = value.rule;
       const resolution = value.resolution;
       const kbaDetail = Object.keys(kbaDetails).length
         ? kbaDetails.filter((article) => article.id === value.rule.node_id)[0]
         : {};
-      const entity = rows.filter(
-        (rowVal, rowKey) =>
-          rowKey % 2 === 0 && rowVal.rule.rule_id === rule.rule_id && rowVal
-      );
```
[the remark this answers is not in the document]
Comment (johnsonm325) [reply src/SmartComponents/SystemAdvisor/SystemAdvisor.js:177]: Okay, I see the issue. `entity` is already defined as the system that all of the recommendations are related to. Here, a new variable named `entity` was set and it filters the rows.

Replace with something like this. It seems to work for me:
```
     const match = rows.find((row) => row?.rule?.rule_id === rule.rule_id);
     const selected = match?.selected;
     const isOpen = match?.isOpen || false;
```

[the remark this answers is not in the document]
Comment (mkholjuraev) [reply src/SmartComponents/SystemAdvisor/SystemAdvisor.js:177]: Nice catch. I have applied your changes and seems to work as expected.

**Review by johnsonm325 — APPROVED** [review]
Nice job!

**Pull request merged**